internal/services/serviceprincipals/synchronization_provision_on_demand_resource.go

```diff
@@ -0,0 +1,140 @@
+	return &schema.Resource{
+		CreateContext: synchronizationProvisionOnDemandResourceCreate,
+		ReadContext:   synchronizationProvisionOnDemandResourceRead,
+		DeleteContext: synchronizationProvisionOnDemandResourceDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(15 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+		SchemaVersion: 0,
+
+		Schema: map[string]*schema.Schema{
+			"service_principal_id": {
+				Description:      "The object ID of the service principal for which this synchronization job should be provisioned",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validate.UUID,
+			},
+			"job_id": {
+				Description: "The identifier for the synchronization jop.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"parameters": {
+				Description: "Represents the objects that will be provisioned and the synchronization rules executed. The resource is primarily used for on-demand provisioning.",
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"rule_id": {
+							Description: "The identifier of the synchronizationRule to be applied. This rule ID is defined in the schema for a given synchronization job or template.",
+							Type:        schema.TypeString,
+							Required:    true,
+							ForceNew:    true,
+						},
+						"subjects": {
+							Description: "The identifiers of one or more objects to which a synchronizationJob is to be applied.",
+							Type:        schema.TypeList,
+							Required:    true,
+							ForceNew:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"object_id": {
+										Description: "The identifier of an object to which a synchronizationJob is to be applied. Can be one of the following:\nAn onPremisesDistinguishedName for synchronization from Active Directory to Azure AD.\nThe user ID for synchronization from Azure AD to a third-party.\nThe Worker ID of the Workday worker for synchronization from Workday to either Active Directory or Azure AD.",
+										Type:        schema.TypeString,
+										Required:    true,
+									},
+									"object_type_name": {
+										Description: "The type of the object to which a synchronizationJob is to be applied. Can be one of the following:\nuser for synchronizing between Active Directory and Azure AD.\nUser for synchronizing a user between Azure AD and a third-party application.\nWorker for synchronization a user between Workday and either Active Directory or Azure AD.\nGroup for synchronizing a group between Azure AD and a third-party application.",
+										Type:        schema.TypeString,
+										Required:    true,
```

Comment:
If these values are known, we should validate them:

```suggestion
										Required:    true,
										ValidateFunc: validation.StringInSlice([]string{
											"Group",
											"user",
											"User",
											"Worker".
										},
```